addons/account/models/account_reconcile_model.py

```diff
@@ -802,26 +802,36 @@ def _get_invoice_matching_rule_result(self, st_line, candidates, aml_ids_to_excl
         new_treated_aml_ids = set()
         candidates, priorities = self._filter_candidates(candidates, aml_ids_to_exclude, reconciled_amls_ids)
 
-        # Special case: the amounts are the same, submit the line directly.
         st_line_currency = st_line.foreign_currency_id or st_line.currency_id
         candidate_currencies = set(candidate['aml_currency_id'] or st_line.company_id.currency_id.id for candidate in candidates)
+        kept_candidates = candidates
         if candidate_currencies == {st_line_currency.id}:
+            kept_candidates = []
+            sum_kept_candidates = 0
             for candidate in candidates:
-                residual_amount = candidate['aml_currency_id'] and candidate['aml_amount_residual_currency'] or candidate['aml_amount_residual']
-                if st_line_currency.is_zero(residual_amount + st_line.amount_residual):
-                    candidates, priorities = self._filter_candidates([candidate], aml_ids_to_exclude, reconciled_amls_ids)
+                candidate_residual = candidate['aml_currency_id'] and candidate['aml_amount_residual_currency'] or candidate['aml_amount_residual']
```

Comment:
In common cases, if the residual amount in foreign currency reaches 0 and you still have a residual amount in domestic currency, an exchange difference move should be created at reconciliation to bring everything down to 0. 

We check everything has the same currency in this specific block, so no multicurrencies issues with it ... BUT there might be an issue in case some candidate is already reconciled with something that's not in its currency, and that's bringing amount_residual_currency to 0 ve though the reconciliation isn't full in domestic currency, I guess. @smetl would be a better choice than me to discuss that, when he's back.

Reply:
> BUT there might be an issue in case some candidate is already reconciled with something that's not in its currency, and that's bringing amount_residual_currency to 0 ve though the reconciliation isn't full in domestic currency

Yes, that is what I was worried about. I feel like just using a ternary condition would settle the debate.

Reply:
Agreed. Will do.

Reply:
Now that I think about it again, isn't the currency always set on move lines? Even if it is in the company's currency?

Reply:
Indeed. The candidate dict are directly generated from the move lines through SQL, so it should be alright just using amount_residual_currency.

Reply:
@william-andre Done :)